src/components/subtitlesync/subtitlesync.js

```diff
@@ -1,170 +1,225 @@
+
+        textField.updateOffset = (offset) => {
+            textField.textContent = offset + 's';
+        };
+
+        textField.addEventListener('click', () => {
+            // keep focus to prevent fade with osd
+            textField.hasFocus = true;
+        });
+
+        textField.addEventListener('keydown', (event) => {
+            if (event.key === 'Enter') {
+                // if input key is enter search for float pattern
+                let inputOffset = /[-+]?\d+\.?\d*/g.exec(textField.textContent);
+                if (inputOffset) {
+                    inputOffset = parseFloat(inputOffset[0]);
+                    this.slider.updateOffset(inputOffset);
+                } else {
+                    textField.textContent = (playbackManager.getPlayerSubtitleOffset(this.player) || 0) + 's';
+                }
+                textField.hasFocus = false;
+                event.preventDefault();
+            } else {
+                // keep focus to prevent fade with osd
+                textField.hasFocus = true;
+                if (event.key.match(/[+-\d.s]/) === null) {
+                    event.preventDefault();
+                }
+            }
+            // FIXME: TV layout will require special handling for navigation keys. But now field is not focusable
+            event.stopPropagation();
+        });
+
+        textField.blur = function() {
+            // prevent textfield to blur while element has focus
+            if (!this.hasFocus && this.prototype) {
+                this.prototype.blur();
+            }
+        };
     }
 
-    subtitleSyncSlider.updateOffset = function (sliderValue) {
-        // default value is 0s = 0ms
-        this.value = sliderValue === undefined ? 0 : sliderValue;
+    updateOffset() {
+        const value = parseFloat(this.slider.value);
 
-        updateSubtitleOffset();
-    };
+        // set new offset
+        playbackManager.setSubtitleOffset(value, this.player);
 
-    subtitleSyncSlider.addEventListener('change', () => updateSubtitleOffset());
+        // synchronize with textField value
+        this.textField.updateOffset(value);
 
-    subtitleSyncSlider.getBubbleHtml = function (_, value) {
-        return '<h1 class="sliderBubbleText">'
-            + (value > 0 ? '+' : '') + parseFloat(value) + 's'
-            + '</h1>';
-    };
+        // update current offset
+        this.currentOffset = value;
+    }
 
-    subtitleSyncCloseButton.addEventListener('click', function () {
-        playbackManager.disableShowingSubtitleOffset(player);
-        SubtitleSync.prototype.toggle('forceToHide');
-    });
+    adjustOffset(delta) {
+        const value = parseFloat(this.slider.value) + delta;
+        this.slider.updateOffset(value);
+    }
 
-    instance.element = parent;
+    setOffset(offset) {
+        this.currentOffset = offset;
+        this.slider.value = offset.toString();
+        this.textField.updateOffset(offset);
+    }
```

Comment:
Alternatively (to `currentOffset` and `setOffset`), we can declare a property with a custom getter and setter.
```js
get currentOffset() {
    return parseFloat(this.slider.value);
}

set currentOffset(offset) {
    this.slider.value = offset.toString();
    this.textField.updateOffset(offset);
}
```

Reply:
Good call, thanks! I also added the playback manager offset call into it

Reply:
```
        slider.addEventListener('change', () => {
            // eslint-disable-next-line no-self-assign
            this.currentOffset = this.currentOffset;
        });
```

not a JS native, but I wanted the slider change to trigger setting of the offset so I am assigning the value to itself, is there a better way to do this, or am I missing something

Reply:
> not a JS native, but I wanted the slider change to trigger setting of the offset so I am assigning the value to itself, is there a better way to do this, or am I missing something (would assigning to `parseFloat(this.slider.value)` be more desirable?)

I'm fine with self-assigning approach. Alternatively, you can extract the playbackManager and textField updates in a separate function, as it was before: https://github.com/jellyfin/jellyfin-web/blob/fec58cc8aaa9dc665105d2677b5b721415360aee/src/components/subtitlesync/subtitlesync.js#L75-L81.